src/Microsoft.NetCore.Analyzers/Core/Security/DoNotDisableCertificateValidation.cs

```diff
@@ -0,0 +1,162 @@
+                            WellKnownTypes.SystemNetSecurityRemoteCertificateValidationCallback);
+                    if (systemNetSecurityRemoteCertificateValidationCallbackTypeSymbol == null)
+                    {
+                        return;
+                    }
+                    
+                    compilationStartAnalysisContext.RegisterOperationAction(
+                        (OperationAnalysisContext operationAnalysisContext) =>
+                        {
+                            IDelegateCreationOperation delegateCreationOperation =
+                                (IDelegateCreationOperation)operationAnalysisContext.Operation;
+                            
+                            if (delegateCreationOperation.Type == systemNetSecurityRemoteCertificateValidationCallbackTypeSymbol)
+                            {
+                                bool flag = true;
+
+                                var kind = delegateCreationOperation.Target.Kind;
+                                switch (kind)
+                                {
+                                    case OperationKind.AnonymousFunction:
+                                        flag = dealWithAnonmousFunction(delegateCreationOperation.Target);
+                                        break;
+
+                                    case OperationKind.MethodReference:
+                                        IMethodReferenceOperation methodReferenceOperation = (IMethodReferenceOperation)delegateCreationOperation.Target;
+                                        flag = dealWithMethodReference(methodReferenceOperation);
+                                        break;
+                                }
+
+                                if (!flag)
+                                    operationAnalysisContext.ReportDiagnostic(
+                                        Diagnostic.Create(
+                                            Rule,
+                                            delegateCreationOperation.Syntax.GetLocation(),
+                                            kind.ToString()));
+                            }
+                        },
+                        OperationKind.DelegateCreation);
+                });
+        }
+
+        private static bool dealWithAnonmousFunction(IOperation operation)
+        {
+            bool flag = false;
+
+            var descendants = operation.Descendants();
+            foreach (var tmp in descendants)
+            {
+                if (tmp.Kind == OperationKind.Return)
+                {
+                    IReturnOperation a=(IReturnOperation)tmp;
+                    bool returnedValue = (bool)a.ReturnedValue.ConstantValue.Value;
+                    if (returnedValue == false)
+                    {
+                        flag = true;
+                        break;
+                    }
+                }
+            }
+
+            return flag;
```

Comment:
Is this method trying to find if there is any `return false;` within the method body?